gguf-py/gguf/gguf_writer.py

```diff
@@ -247,7 +301,14 @@ def add_tensor_info(
             if tensor_dtype == np.uint8:
                 tensor_shape = quant_shape_from_byte_shape(tensor_shape, raw_dtype)
 
-        self.tensors[name] = TensorInfo(shape=tensor_shape, dtype=dtype, nbytes=tensor_nbytes)
+        # split when over tensor limit
+        if (self.split_max_tensors != 0 and len(self.tensors[-1]) >= self.split_max_tensors \
+                # or split when over size limit
+                or self.split_max_size != 0 and sum(ti.nbytes for ti in self.tensors[-1].values()) + tensor_nbytes > self.split_max_size):
```

Comment:
We can do that. Because we write the tensor to the shard immediately after, this only affects the very first tensor... but because `token_embd.weight` is often the first and one of the biggest tensors, that's one of the most common cases.